### PR TITLE
model: Add obsolete_tag to transfer_t

### LINKED ISSUE #838: model: Tag changes after bundle normalization
In `transfer_iterator_new()`, `transfer_ctx_hash()` calculates bundle hash and normalize bundle hash by incrementing `transfers[0]->tag`. 
```
// In transfer_ctx_hash()
// normalize
    normalize_flex_hash(transfer_ctx->bundle, normalized_hash);
    // checking 'M'
    for (i = 0; i < HASH_LENGTH_TRYTE; i++) {
      if (normalized_hash[i] == 13) {
        // Insecure bundle. Increment Tag and recompute bundle hash.
        add_assign(first_tx_tag_trits, NUM_TRITS_TAG, 1);
        flex_trits_from_trits((flex_trit_t*)transfers[0]->tag, NUM_TRITS_TAG,
                              first_tx_tag_trits, NUM_TRITS_TAG, NUM_TRITS_TAG);
        goto loop;
      }
}
```

But here we do not preserve the original `tag` value, transaction `tag` is incorrect after `transfer_iterator_next()`.
Since both `tag` and `obsolete_tag` are copies from `transfer->tag`. 
```
// transfer_iterator_next()
transaction_set_tag(transaction, transfer->tag);
transaction_set_obsolete_tag(transaction, transfer->tag);
```

Maybe we should create a new variable to record `obsolete_tag`.
reference: [iota.js finalizeBundle](https://github.com/iotaledger/iota.js/blob/d86074f2840c17c72c9d0a55ded4a9b8d0998698/packages/bundle/src/index.ts#L141)

### PR DESCRIPTION
**Pull request by jkrvivian**
By separating obsolete_tag and tag in transfer_t, obsolete_tag will be used in bundle normalization while tag will remain the same as input.

tag and obsolete_tag checks are added to `test_transfer.c`

Close #838 

# Test Plan:
Test `common/model/tests/test_transfer.c`

### CODE REVIEW

**Review by tsvisabo — DISMISSED** [review]

File: common/model/transfer.c [hunk 1]
```diff
@@ -167,6 +167,8 @@ transfer_t* transfer_data_new(flex_trit_t const* const address,
     transfer->address = address;
     transfer->value = 0;
     transfer->tag = tag;
+    flex_trits_slice(transfer->obsolete_tag, NUM_TRITS_TAG, tag, NUM_TRITS_TAG,
```
Comment (thibault-martinez) [inline common/model/transfer.c]: `tag` and `obsolete_tag` have the same  size, please use `memcpy`. `flex_trits_slice` is really inefficient.

File: common/model/transfer.c [hunk 2]
```diff
@@ -213,6 +215,8 @@ transfer_t* transfer_value_out_new(transfer_value_out_t const* const output,
   tf->address = address;
   tf->value = value;
   tf->tag = tag;
+  flex_trits_slice(tf->obsolete_tag, NUM_TRITS_TAG, tag, NUM_TRITS_TAG, 0,
```
Comment (thibault-martinez) [inline common/model/transfer.c]: Ditto

File: common/model/transfer.c [hunk 3]
```diff
@@ -260,6 +264,8 @@ transfer_t* transfer_value_in_new(flex_trit_t const* const address,
   tf->type = VALUE_IN;
   tf->address = address;
   tf->tag = tag;
+  flex_trits_slice(tf->obsolete_tag, NUM_TRITS_TAG, tag, NUM_TRITS_TAG, 0,
```
Comment (thibault-martinez) [inline common/model/transfer.c]: Ditto

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Comment by thibault-martinez** [comment]
Thanks !

**Pull request merged**